app-creator/client/src/utils/template-generation.ts

```diff
@@ -0,0 +1,120 @@
+/**
+ * @fileoverview this file contains the logic for rendering a template given its corresponding JSON string. It is used
+ * whenever we want to display a new template on the story-board.
+ */
+import { AppCreatorStore, WidgetMetaData } from '../redux/reducer';
+import { ROOT_ID } from './constants';
+import { store } from '../redux/store';
+import { setSelectedTemplate } from '../redux/actions';
+import { Dropzone } from '../widgets/dropzone-widget/dropzone-widget';
+import { DraggableWidget } from '../widgets/draggable-widget/draggable-widget';
+import { getIdPrefix } from './helpers';
+import { EEWidget } from '../redux/types/types';
+import { WidgetType } from '../redux/types/enums';
+import { Panel } from '../widgets/ui-panel/ui-panel';
+import { Map } from '../widgets/ui-map/ui-map';
+
+export function generateUI(
+  template: AppCreatorStore['template'],
+  node: HTMLElement
+) {
+  const templateCopy = Object.assign({}, template);
+
+  function helper(widgetData: WidgetMetaData): HTMLElement {
```

Comment:
Perhaps replace `helper` with a explicit function name, so we know what it helps to do?